File: app/src/main/java/com/draabek/fractal/gl/GLSLFractal.java
package com.draabek.fractal.gl;

import androidx.annotation.NonNull;

import com.draabek.fractal.fractal.Fractal;
import com.draabek.fractal.fractal.FractalViewWrapper;

public class GLSLFractal extends Fractal {
    /**
     * A fractal subclass rendered using the GLSL language
     * directly on the graphics card
     */
    private String[] shaders = null;

    public GLSLFractal() {
        super();
    }

    @Override
    public Class<? extends FractalViewWrapper> getViewWrapper() {
        return RenderImageView.class;
    }

    public String[] getShaders() {
        return shaders;
    }

    public void setShaders(@NonNull String[] shaders) {
        this.shaders = shaders;
    }

}

File: app/src/main/java/com/draabek/fractal/gl/GLSLFractal.kt
package com.draabek.fractal.gl

import com.draabek.fractal.fractal.Fractal
import com.draabek.fractal.fractal.FractalViewWrapper

class GLSLFractal : Fractal() {
    /**
     * A fractal subclass rendered using the GLSL language
     * directly on the graphics card
     */
    var shaders: Array<String>? = null

    override val viewWrapper: Class<out FractalViewWrapper>
        get() = RenderImageView::class.java

}
